quasar-core/src/main/java/co/paralleluniverse/concurrent/util/ThreadAccess.java

```diff
@@ -51,7 +50,17 @@
         try {
             MethodHandles.Lookup l = MethodHandles.lookup();
             l = MethodHandles.privateLookupIn(Thread.class, l);
-            TARGET = l.findVarHandle(Thread.class, "target", Runnable.class); // l.unreflectVarHandle(Thread.class.getDeclaredField("target"));
+            {
+                // ENT-5489, try name "target" first then try name "runnable".
+                // runnable is the name used in openj9.
+                VarHandle target;
+                try {
+                    target = l.findVarHandle(Thread.class, "target", Runnable.class);
+                } catch (NoSuchFieldException ex) {
```

Comment:
Is it possible to check your running under openJ9 and then check for this, rather than relying on exception. Am thinking this could hit performance?

Reply:
This is in a static initializer, so is only done once in process.

Reply:
OK - not a perf issue then